R/module_nested_tabs.R

```diff
@@ -364,11 +364,11 @@ srv_teal_module.teal_module <- function(id,
 .resolve_module_datanames <- function(data, modules) {
   stopifnot("data_rv must be teal_data object." = inherits(data, "teal_data"))
   if (is.null(modules$datanames) || identical(modules$datanames, "all")) {
-    .topologically_sort_datanames(ls(teal.code::get_env(data)), teal.data::join_keys(data))
+    names(data) # TODO: Reviewer: Check that previous code does nothing.
   } else {
     intersect(
-      .include_parent_datanames(modules$datanames, teal.data::join_keys(data)),
-      ls(teal.code::get_env(data))
+      names(data), # Keep topological order from teal.data::names()
+      .include_parent_datanames(modules$datanames, teal.data::join_keys(data))
```

Comment:
This would solve the problem of `datanames = ".iris"`, but even if we included dot-prefixed-dataset in module's `data` it will be hidden when using `names` in the module. I think your proposition is relevant and consistent - we ignore all dot-prefixed-datasets and that's it, period.

```r
    intersect(
      .include_parent_datanames(modules$datanames, teal.data::join_keys(data)),
      ls(data)
    )
```

Reply:
So do you want to keep `names()` for now and rely on the topological order determined in `teal.data::names.teal_data` or use `ls()` here and order it with `.include_parent_....`?